jetty-util/src/main/java/org/eclipse/jetty/util/Scanner.java

```diff
@@ -607,27 +619,49 @@ public void removeListener(Listener listener)
     public void doStart() throws Exception
     {
         if (LOG.isDebugEnabled())
-            LOG.debug("Scanner start: scanInStart={}, reportExists={}, depth={}, rprtDirs={}, interval={}, filter={}, scannables={}",
-                      _scanInStart, _reportExisting, _scanDepth, _reportDirs, _scanInterval, _filter, _scannables);
+            LOG.debug("Scanner start: deferScan={}, reportExists={}, depth={}, rprtDirs={}, interval={}, filter={}, scannables={}",
+                _deferInitialScan, _reportExisting, _scanDepth, _reportDirs, _scanInterval, _filter, _scannables);
 
-        if (_scanInStart)
+        if (!_deferInitialScan)
         {
-            if (_reportExisting)
-            {
-                // if files exist at startup, report them
-                scan();
-                scan(); // scan twice so files reported as stable
-            }
-            else
-            {
-                //just register the list of existing files and only report changes
-                _prevScan = scanFiles();
-            }
+            _scheduler.start();
+            startup();
         }
 
         super.doStart();
+    }
+
+    /**
+     * Perform the initial scan of the directories {@link #setScanDirs(List),
+     * and start the scan interval schedule.
+     */
+    public void startup()
```

Comment:
```suggestion
    public void activateScan()
```

Reply:
I had it as `initialScan()` for a bit, then realized it felt odd to document and use with that name.
The `startup()` name fit better in both documentation and usage.

Using `activateScan()` is awkward, but for a different reason, it is indistinguishable from just `.scan()` which is also in the API.

Reply:
it's really more of a `startScanning()` type of method.
But then you would assume a `stopScanning()` is also present.

I'm not a fan of `activateScan` or `scanActivated` as names.

Reply:
Perhaps `beingMonitoring()` and `stopMonitoring()` ?

Reply:
Since we stop the scheduler task in `doStop()` and and have a `doStart()` which could start the scanning, how about we call it simply `doStartScanning()` or `.doScanning()` to be in sync with the other naming?
If someone wants to stop the scanner, then just call `.stop()` (like normal)
If they defer the initial scan, then they have to call `.doStartScanning()` (or `.doScanning()`) ?

Reply:
Other options ...

```
.initiateScanning()
.beginScanning()
.doScanning()
.startScanning()
```

I can even factor out the current `.doStop()` logic to cancel the scheduled task into a new method like `.stopScanning()` to balance out the naming. (the new `.doStop()` would just call this new `.stopScanning()` method)